florist/api/server.py

```diff
@@ -0,0 +1,118 @@
+"""FLorist server FastAPI endpoints."""
+import logging
+from typing import List
+
+import requests
+from fastapi import FastAPI, Form
+from fastapi.responses import JSONResponse
+from typing_extensions import Annotated
+
+from florist.api.monitoring.metrics import wait_for_metric
+from florist.api.servers.common import ClientInfo, ClientInfoParseError, Model
+from florist.api.servers.launch import launch_local_server
+
+
+app = FastAPI()
+LOGGER = logging.getLogger("uvicorn.error")
+
+START_CLIENT_API = "api/client/start"
+
+
+@app.post("/api/server/start_training")
+def start_training(
+    model: Annotated[str, Form()],
+    server_address: Annotated[str, Form()],
+    n_server_rounds: Annotated[int, Form()],
+    batch_size: Annotated[int, Form()],
+    local_epochs: Annotated[int, Form()],
+    redis_host: Annotated[str, Form()],
+    redis_port: Annotated[str, Form()],
+    clients_info: Annotated[str, Form()],
+) -> JSONResponse:
+    """
+    Start FL training by starting a FL server and its clients.
+
+    Should be called with a POST request and the parameters should be contained in the request's form.
+
+    :param model: (str) The name of the model to train. Should be one of the values in the enum
+        florist.api.servers.common.Model
+    :param server_address: (str) The address of the FL server to be started. It should be comprised of
+        the host name and port separated by colon (e.g. "localhost:8080")
+    :param n_server_rounds: (int) The number of rounds the FL server should run.
+    :param batch_size: (int) The size of the batch for training
```

Comment:
nit: It seems most of the time we are ending the argument definitions with periods so we mine as well stay consistent